src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,80 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => void,
+};
+
+export const Pagination: React.FC<Props> = ({
+  total, perPage, currentPage, onPageChange,
+}) => {
+  const pageCount = Math.ceil(total / perPage);
+  const pages = getNumbers(1, pageCount);
+  const firstPage = currentPage === 1;
+
+  return (
+    <ul className="pagination">
+      <li
+        className={classNames('page-item', { disabled: firstPage })}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={firstPage}
+          onClick={() => {
+            if (firstPage) {
+              return;
+            }
+
+            onPageChange(currentPage - 1);
+          }}
+        >
+          «
+        </a>
+      </li>
+
+      {pages.map(page => (
+        <li
+          className={classNames('page-item', { active: currentPage === page })}
+          key={page}
+        >
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${page}`}
+            onClick={() => onPageChange(page)}
+          >
+            {page}
+          </a>
+        </li>
+      ))}
+
+      <li
+        className={classNames('page-item', {
+          disabled: currentPage === pageCount,
+        })}
+      >
+        <a
+          data-cy="nextLink"
+          className="page-link"
+          href="#next"
+          aria-disabled={currentPage === pageCount}
+          onClick={() => {
+            if (currentPage === pageCount) {
```

Comment:
makeVariable for this condition and use in all places
`const isLastPage = currentPage === pageCount`